src/coreComponents/common/MpiWrapper.hpp

```diff
@@ -1115,7 +1125,32 @@ void MpiWrapper::reduce( Span< T const > const src, Span< T > const dst, Reducti
   reduce( src.data(), dst.data(), LvArray::integerConversion< int >( src.size() ), getMpiOp( op ), root, comm );
 }
 
+// Mpi helper function to return  struct containing the max value and location across ranks
+template< typename T >
+T MpiWrapper::maxValLoc( T localValueLocation, MPI_Comm comm )
```

Comment:
There's certainly a more efficient way to do this, but my guess is this isn't in a performance critical section so it doesn't matter.